cinema/views.py

```diff
@@ -56,3 +89,65 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        if movie := self.request.query_params.get("movie"):
+            queryset = queryset.filter(movie__id=int(movie))
+
+        if date := self.request.query_params.get("date"):
+            queryset = queryset.filter(show_time__date=date)
+
+        if self.action == "list":
+            queryset = (
```

Comment:
i think u can simplify this point use select_related instead of double prefetch related 👯